crates/subgraph/src/apy.rs

```diff
@@ -0,0 +1,777 @@
+                            && v.input_vault_balance_change.vault.vault_id.0 == input.id
+                            && v.output_vault_balance_change.vault.vault_id.0 == output.id
+                    })
+                    .collect::<Vec<&Trade>>();
+
+                // calculate the pair ratio (in amount/out amount)
+                let ratio = if pair_trades.is_empty() {
+                    None
+                } else {
+                    // convert input and output amounts to 18 decimals point
+                    // and then calculate the pair ratio
+                    let input_amount = to_18_decimals(
+                        ParseUnits::U256(
+                            U256::from_str(&pair_trades[0].input_vault_balance_change.amount.0)
+                                .unwrap(),
+                        ),
+                        pair_trades[0]
+                            .input_vault_balance_change
+                            .vault
+                            .token
+                            .decimals
+                            .as_ref()
+                            .map(|v| v.0.as_str())
+                            .unwrap_or("18"),
+                    );
+                    let output_amount = to_18_decimals(
+                        ParseUnits::U256(
+                            U256::from_str(
+                                &pair_trades[0].output_vault_balance_change.amount.0[1..],
+                            )
+                            .unwrap(),
+                        ),
+                        pair_trades[0]
+                            .output_vault_balance_change
+                            .vault
+                            .token
+                            .decimals
+                            .as_ref()
+                            .map(|v| v.0.as_str())
+                            .unwrap_or("18"),
+                    );
+                    #[allow(clippy::unnecessary_unwrap)]
+                    if input_amount.is_err() || output_amount.is_err() {
+                        None
+                    } else {
+                        Some(
+                            input_amount
+                                .unwrap()
+                                .get_signed()
+                                .saturating_mul(one)
+                                .checked_div(output_amount.unwrap().get_signed())
+                                .unwrap_or(I256::MAX),
+                        )
+                    }
+                };
+                pair_ratio_map.insert(
+                    TokenPair {
+                        input: input.token.clone(),
+                        output: output.token.clone(),
+                    },
+                    ratio,
+                );
+            }
+        }
+    }
+    pair_ratio_map
+}
+
+/// Converts a U256 or I256 to a fixed point U256 or I256 given the decimals point
+pub fn to_18_decimals<T: TryInto<Unit, Error = UnitsError>>(
```

Comment:
we don't have a fn for this already?

Reply:
not as far as I checked

Reply:
ok, well this doesn't seem like a good place for this function if anyone else is going to discover it and use it